src/routes/hydra.ts

```diff
@@ -1,162 +1,209 @@
-              .then( ({body, response} ) => {
-                // We need to get the email of the user. We don't want to do that via traits as
-                // they are dynamic. They would be part of the PublicAPI. That's not true
-                // for identity.addresses So let's get it via the AdmninAPI
-                const subject = body.identity.id
-                kratosAdminEndpoint
-                  .getIdentity(body.identity.id) 
-                  .then( ({response}) => {
-                    // User is authenticated, accept the LoginRequest and tell Hydra
-                    let acceptLoginRequest : AcceptLoginRequest = new AcceptLoginRequest()
-                    // We should probably use "id" here but the gitlab-scenario only works via "email"
-                    acceptLoginRequest.subject = subject
-                    acceptLoginRequest.context = (response as any).body
-   
-                    //acceptLoginRequest.subject = (body.identity.traits as { email: string }).email || "fallback@mail.com"
-                    
-                    return hydraAdminEndpoint.acceptLoginRequest(challenge, acceptLoginRequest)
-                      .then((hydraResponse: any) => {
-                        // All we need to do now is to redirect the user back to hydra!
-                        res.redirect(hydraResponse.body.redirectTo)
-                    })
-                  })
-
-              })
-          } else {
-            logger.error("Request and challenge are both set but user is not authenticated. Unknown state!")
-            res.status(400).send('Request and challenge are bot set but user is not authenticated. Please try again!');
-            next()
-          }
-        })
-        .catch((err) => {
-          // Something went wrong with validating hydra's challenge getting the LoginRequest
-          logger.error(err)
-          next()
-        });
-    }
+    logger.debug('hydraChallenge exists: ' + hydraChallenge)
+  }
+
+  if (config.securityMode !== SECURITY_MODE_STANDALONE) {
+    next(
+      new Error(
+        'Interaction with ORY Hydra only works in security standalone mode right now.'
+      )
+    )
+    return
+  }
+  const kratosSessionCookie = req.cookies.ory_kratos_session
+
+  if (kratosRequest) {
+    // FIXME @k9ert this state is undefined.
+    next(new Error('what should be done here?'))
+    return
+  }
+
+  if (!kratosSessionCookie) {
+    // 3. Initiate login flow with Kratos
+    // prompt=login forces a new login from kratos regardless of browser sessions - this is important because we are letting Hydra handle sessions
+    // redirect_to ensures that when we redirect back to this url, we will have both the initial hydra hydraChallenge and the kratos kratosRequest id in query params
+    logger.info(
+      'Initiating ORY Kratos Login flow because neither a ORY Kratos Login Request nor a valid ORY Kratos Session was found.'
+    )
+
+    // FIXME @k9ert this will not work behind a proxy
```

Comment:
```
const returnTo = encodeURI(`${req.protocol}://${req.headers.host}${req.url}`)
```

will be something like `http://kratos-app.internal.svc.local:1234/asdf` behind a proxy. It probably makes sense to use `config.baseUrl` instead (from README):

> - `BASE_URL` (optional): The base url of this app. If served e.g. behind a proxy or via
  GitHub pages this would be the path, e.g. `https://mywebsite.com/kratos-selfservice-ui-node/`.
  **Must be absolute!**